V-03 / F-02 / F-05 Выполнены

Feintty opened the pull request:
V-03 Созданы стили для checkbox
V-03 Сверстана страница допов
F-02 Реализовано переключение вкладок
F-02 Реализован сброс последующих вкладок при смене значений
F-05 Реализовано заполнение заказа дополнительными данными
F-05 Введено ограничение на вид тарифа в зависимости от длительности аренды

Feintty committed:
Merge pull request #10 from Feintty/review

Review

Feintty committed:
V-03 Созданы стили для checkbox
V-03 Сверстана страница допов
F-02 Реализовано переключение вкладок
F-02 Реализован сброс последующих вкладок при смене значений
F-05 Реализовано заполнение заказа дополнительными данными
F-05 Введено ограничение на вид тарифа в зависимости от длительности аренды

konkonskiy commented on need-for-drive/src/components/AdditionsTab/AdditionsTab.js:
```diff
@@ -0,0 +1,226 @@
+  const [secondDate, setSecondDate] = useState(null)
+  const [error, setError] = useState()
+
+  const adds = [
```
в отдельный файл

konkonskiy commented on need-for-drive/src/components/AdditionsTab/AdditionsTab.js:
```diff
@@ -0,0 +1,226 @@
+  time: PropTypes.objectOf()
+}
+
+export default AdditionsTab
```
Дроби компоненту на более мелкие.

konkonskiy commented on need-for-drive/src/components/AdditionsTab/AdditionsTabApi.js:
```diff
@@ -0,0 +1,24 @@
+const mainHeaders = {
+  headers: {
```
Разве у тебя не одинаковые хедеры относительно предыдущих запросов?

konkonskiy commented on need-for-drive/src/components/CarsTab/CarsTab.js:
```diff
@@ -65,8 +76,11 @@ const CarsTab = ({ carToOrder }) => {
         <div className="cars-tab__table">
           {filterCars(cars).map((el) => (
             <CarCard
+              id={el.id}
```
key не потерял?

konkonskiy commented on need-for-drive/src/components/DateTimePicker/DateTimePicker.scss:
```diff
@@ -0,0 +1,28 @@
+.data-time {
+  @include flex-row;
+  position: relative;
+  .data-time__description {
```
&__description {

konkonskiy commented on need-for-drive/src/components/DateTimePicker/DateTimePicker.scss:
```diff
@@ -0,0 +1,28 @@
+    padding: 3px 10px;
+  }
+
+  .data-time__close {
```
&__close  {

konkonskiy commented on need-for-drive/src/components/LocationTab/LocationTab.js:
```diff
@@ -27,8 +27,10 @@ const LocationTab = ({ cityAndPointToOrder }) => {
         if (point.cityId !== null) {
-          if (point.cityId.name === city) {
-            return point
+          if (!/[0-9]|Город/.test(point.cityId.name)) {
```
!/[0-9]|Город/ в константу

konkonskiy commented on need-for-drive/src/pages/OrderPage/OrderPage.js:
```diff
@@ -80,23 +151,47 @@ const OrderPage = () => {
+            rightHand={additionsData.isRighthand}
           />
         </div>
       </div>
```
Подумай как сократить компоненту

Feintty committed:
F-05 Вкладка с допами разбита на подкомпоненты
F-05 Логика связанная с заполнением заказа вынесена в отдельный элемент
F-05 Реализовано уведомление пользователя об ограничениях цены
F-03 Изменено центрирование карты

konkonskiy commented on need-for-drive/src/components/AdditionsTab/AdditionsTabCheckboxes.js:
```diff
@@ -0,0 +1,63 @@
+    }
+  }
+
+  const defaultCheckbox = (value) => {
```
Функции обычно называют глаголами. В твоем случае это isDefaultCheckbox

konkonskiy commented on need-for-drive/src/components/AdditionsTab/AdditionsTabCheckboxes.js:
```diff
@@ -0,0 +1,63 @@
+      </label>
+    ))
+
+  return <>{createAddsCheckboxes()}</>
```
Вот это очень странная штука 
Что тебе помешало сделать так 
```
return (<>
{adds.map((add) => (
      <label className="checkbox-container" htmlFor onChange={checkboxOnChange}>
        <input
          className="checkbox-input"
          type="checkbox"
          name="adds"
          value={add}
          checked={defaultCheckbox(add)}
        />
        <span className="checkbox-text">{add}</span>
      </label>
    )}
</>)
```
Зачем засорять память лишней функцией?

konkonskiy commented on need-for-drive/src/components/AdditionsTab/AdditionsTabColorRadios.js:
```diff
@@ -0,0 +1,37 @@
+      </label>
+    ))
+
+  return <>{createColorRadios()}</>
```
тоже самое

Feintty committed:
F-05 Исправлен баг с датами
F-05 Сокращены избыточные функции

Feintty committed:
F-05 Исправлен сброс даты при одинаковых значениях
